### ISSUE TITLE
Some Static Groups do have more than one static unit

### ISSUE THREAD

**Issue by NovaLogicDev** [akaAgar/briefing-room-for-dcs#388]
### Brief Description: 
Create an objective, set its task target as structures (any, military, production, offshore) with a Target count of very high.
There is a chance that the static group will be something like Outpost, which is only configured with one building, which is not a very high target count.

### Expected: 
Should get a grouping of units such as a couple of buildings/gas tanks

After debugging, it looks like there is a drop in connection between the configured min,max target count configured for a given ObjectiveTarget ini file such as in Database/ObjectiveTargets/StructureMilitary.ini
If I configure that file to have a VeryHigh min-max of 8,12 when generating the mission only sometimes will I get a group of buildings that actually has more than one building.

This is caused in two part I think. 
1) DBEntryCoalition.GetRandomUnits will modify the unit count to one because it understands correctly that a group of statics in dcs can only be a single static, however this is now a regression with the #288 behavior because although dcs groups are size one, we can place multiple statics at a single objective.

2) This is the reason 1 is important with my limited understanding. During the creation of a Static group in UnitMaker.AddStaticGroup, it has no knowledge of how many buildings the user had requested with the TargetCount field.

On the surface, a simple solution would be to count how many units are in a staticgroup and only spawn that static group if it meets the min-max requirements set by the user.... Trouble with this is that staticgroups have a fixed amount of statics, and this may not exactly match the unit count ask exactly, or no groups may exist within a range.

Probably a good idea would be to have some sort of fuzzy understanding of the TargetCount of a given static group and match that with the users request.
I think something like a min max value that measures how many statics are in the configured groups. and then divides the difference by the amount of possible TargetCount values the user can request (5 currently) giving 5 ranges of static count and grouping the configured static groups would work as a "fuzzy understanding"

EX: the static groups for FARP and Ammo Depos would fall into the VeryHigh target count category

I will probably hack together my hairbrained scheme and if its clean enough I can send in a PR

**Comment by john681611** [comment]
Static units were previously always 1 building no matter the count option. Basically, the count option was ignored. Nothing has changed in that regard. 

As you have identified we now allow for groups using the same trick as the SAM sites use, From the generator's point of view, it's still one unit that is represented by multiple buildings. 

It's worth remembering that not all buildings are 1-2 hit kills so for s flight a small base might use up most of their ordnance. Often the bunkers take more than one large weapon hits to be considered done. 

Maybe the quick solution here is to separate singular static targets and groups of static targets into separate families and target groups.

**Comment by NovaLogicDev** [comment]
I've got a mostly working implementation it seems. What I did was reconfigure the ObjectiveTarget configurations for StructuresAny, Military, Offshore and Production so that the different levels of TargetCount have different ranges, I then modified the generator so that the validUnits list will only contain units that fit those parameters. See #389 

In order to ensure that at least some validUnits are returned I check to make sure the the validUnits list actually ends up containing units after limited by the length of DCSIDs, if not I try to fill the list again and drop the minUnitCount by one each time until I get at least one unit in the list.
